athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -1139,6 +758,8 @@ protected static void writeMapValue(UnionMapWriter writer, Field field, BufferAl
             switch (Types.getMinorTypeForArrowType(type)) {
                 case TIMESTAMPMILLITZ:
                     long dateTimeWithZone;
+                    TimeStampMilliTZWriter timeStampMilliTZWriter = isMapOrStruct ? writer.timeStampMilliTZ(field.getName()) : writer.timeStampMilliTZ();
```

Comment:
I'm wondering if we can unconditionally do:
`writer.timeStampMilliTZ(field.getName())`

Where in the worst case, we just get a `null` for `field.getName()` and it doesn't matter anyway?

However, I could be wrong and `field.getName()` might throw an exception. But if it doesn't, I think can get rid of this extra `isMapOrStruct` parameter and always just use `field.getName()`.

This comment also applies to everywhere else that this parameter is being used.

Reply:
Nevermind about this. There's something weird with the generated code that's causing the subclass's `timeStampMilliTZ(String name)` function even though the base class being used here is FieldWriter that doesn't actually have that exposed (at least not according to the java docs)...